src/LSTM/index.js

```diff
@@ -29,130 +29,131 @@ class LSTM {
-      Object.keys(vars).forEach((key) => {
-        if (key.match(regexCell)) {
-          if (key.match(regexWeights)) {
-            this.model[`Kernel_${key.match(/[0-9]/)[0]}`] = vars[key];
-            this.cellsAmount += 1;
-          } else {
-            this.model[`Bias_${key.match(/[0-9]/)[0]}`] = vars[key];
-          }
-        } else if (key.match(regexFullyConnected)) {
-          if (key.match(regexWeights)) {
-            this.model.fullyConnectedWeights = vars[key];
-          } else {
-            this.model.fullyConnectedBiases = vars[key];
-          }
+    const vars = await reader.getAllVariables();
+    Object.keys(vars).forEach((key) => {
+      if (key.match(regexCell)) {
+        if (key.match(regexWeights)) {
+          this.model[`Kernel_${key.match(/[0-9]/)[0]}`] = vars[key];
+          this.cellsAmount += 1;
+        } else {
+          this.model[`Bias_${key.match(/[0-9]/)[0]}`] = vars[key];
+        }
+      } else if (key.match(regexFullyConnected)) {
+        if (key.match(regexWeights)) {
+          this.model.fullyConnectedWeights = vars[key];
         } else {
-          this.model[key] = vars[key];
+          this.model.fullyConnectedBiases = vars[key];
         }
-      });
-      this.loadVocab(path, callback);
+      } else {
+        this.model[key] = vars[key];
+      }
     });
+    await this.loadVocab(path);
+    callback();
   }
 
-  loadVocab(file, callback) {
-    fetch(`${file}/vocab.json`)
-      .then(response => response.json())
-      .then((json) => {
-        this.vocab = json;
-        this.vocabSize = Object.keys(json).length;
-        this.ready = true;
-        callback();
-      }).catch((error) => {
-        console.error(`There has been a problem loading the vocab: ${error.message}`);
-      });
+  async loadVocab(file, callback = () => {}) {
+    const json = await fetch(`${file}/vocab.json`)
+      .then(response => response.json());
+    this.vocab = json;
+    this.vocabSize = Object.keys(json).length;
+    callback();
   }
 
-  async generate(options, callback) {
+  async generate(options, callback = () => {}) {
     const seed = options.seed || this.defaults.seed;
     const length = +options.length || this.defaults.length;
     const temperature = +options.temperature || this.defaults.temperature;
     const results = [];
 
-    if (this.ready) {
-      const forgetBias = tf.tensor(1.0);
-      const LSTMCells = [];
-      let c = [];
-      let h = [];
-
-      const lstm = (i) => {
-        const cell = (DATA, C, H) =>
-          tf.basicLSTMCell(forgetBias, this.model[`Kernel_${i}`], this.model[`Bias_${i}`], DATA, C, H);
-        return cell;
-      };
-
-      for (let i = 0; i < this.cellsAmount; i += 1) {
-        c.push(tf.zeros([1, this.model[`Bias_${i}`].shape[0] / 4]));
-        h.push(tf.zeros([1, this.model[`Bias_${i}`].shape[0] / 4]));
-        LSTMCells.push(lstm(i));
-      }
+    await this.ready;
+    const forgetBias = tf.tensor(1.0);
```

Comment:
@cvalenzuela here and below the tensors are leaking. Same comment above, I'd add a unit test that asserts generate() doesn't create more tensors than you expect.

Reply:
should we wrap this in `tf.tidy()`?